auth/postgres/repo.go

```diff
@@ -0,0 +1,608 @@
+	pat, err := pr.retrievePATFromDB(ctx, userID, patID)
+	if err != nil {
+		return auth.PAT{}, errors.Wrap(repoerr.ErrViewEntity, err)
+	}
+
+	return pat, nil
+}
+
+func (pr *patRepo) RetrieveAll(ctx context.Context, userID string, pm auth.PATSPageMeta) (auth.PATSPage, error) {
+	q := `
+		SELECT 
+		p.id, p.user_id, p.name, p.description, p.issued_at, p.expires_at,
+		p.updated_at, p.revoked, p.revoked_at
+		FROM pats p WHERE user_id = :user_id
+		ORDER BY issued_at DESC
+		LIMIT :limit OFFSET :offset`
+
+	dbPage := dbPagemeta{
+		Limit:  pm.Limit,
+		Offset: pm.Offset,
+		User:   userID,
+	}
+
+	rows, err := pr.db.NamedQueryContext(ctx, q, dbPage)
+	if err != nil {
+		return auth.PATSPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+	}
+	defer rows.Close()
+
+	var items []auth.PAT
+	for rows.Next() {
+		var pat dbPat
+		if err := rows.StructScan(&pat); err != nil {
+			return auth.PATSPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+		}
+		items = append(items, auth.PAT{
+			ID:          pat.ID,
+			User:        pat.User,
+			Name:        pat.Name,
+			Description: pat.Description,
+			IssuedAt:    pat.IssuedAt,
+			ExpiresAt:   pat.ExpiresAt,
+			UpdatedAt:   pat.UpdatedAt,
+			Revoked:     pat.Revoked,
+			RevokedAt:   pat.RevokedAt,
+		})
+	}
+
+	cq := `SELECT COUNT(*) FROM pats p WHERE user_id = :user_id`
+
+	total, err := postgres.Total(ctx, pr.db, cq, dbPage)
+	if err != nil {
+		return auth.PATSPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+	}
+
+	page := auth.PATSPage{
+		PATS:   items,
+		Total:  total,
+		Offset: pm.Offset,
+		Limit:  pm.Limit,
+	}
+	return page, nil
+}
+
+func (pr *patRepo) RetrieveSecretAndRevokeStatus(ctx context.Context, userID, patID string) (string, bool, bool, error) {
+	q := `
+		SELECT p.secret, p.revoked, p.expires_at 
+		FROM pats p
+		WHERE user_id = $1 AND id = $2`
+
+	rows, err := pr.db.QueryContext(ctx, q, userID, patID)
+	if err != nil {
+		return "", true, true, postgres.HandleError(repoerr.ErrNotFound, err)
+	}
+	defer rows.Close()
+
+	var secret string
+	var revoked bool
+	var expiresAt time.Time
+
+	if !rows.Next() {
+		return "", true, true, repoerr.ErrNotFound
+	}
+
+	if err := rows.Scan(&secret, &revoked, &expiresAt); err != nil {
+		return "", true, true, postgres.HandleError(repoerr.ErrNotFound, err)
+	}
+
+	expired := time.Now().After(expiresAt)
+	return secret, revoked, expired, nil
+}
+
+func (pr *patRepo) UpdateName(ctx context.Context, userID, patID, name string) (auth.PAT, error) {
+	q := `
+		UPDATE pats p
+		SET name = :name, updated_at = :updated_at
+		WHERE user_id = :user_id AND id = :id
+		RETURNING id, user_id, name, description, secret, issued_at, updated_at, expires_at, revoked, revoked_at, last_used_at`
+
+	upm := dbPagemeta{
+		User:      userID,
+		ID:        patID,
+		Name:      name,
+		UpdatedAt: time.Now(),
+	}
+
+	rows, err := pr.db.NamedQueryContext(ctx, q, upm)
+	if err != nil {
+		return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	var pat auth.PAT
```

Comment:
I think you should use dbPAT instead of auth.PAT
```suggestion
	if rows.Next() {
		var pat dbPat

		if err := rows.StructScan(&pat); err != nil {
			return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
		}
		res, err := toAuthPat(pat)
		if err != nil {
			return auth.PAT{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
		}

		return res, nil
	}
	return auth.PAT{}, repoerr.ErrNotFound
```